This is required to enable builds in architectures other than amd64
(x86_64). Currently supports i386/armv6/armv7/aarch64.

The upstream as of now "hardcodes" amd64 (x86_64) as the only build
option.

ARCH_DIR is now set from configure.ac

--- src/Makefile.am.orig	2019-03-11 19:42:59 UTC
+++ src/Makefile.am
@@ -4,7 +4,7 @@ noinst_PROGRAMS = goldilocks_gen_tables
 
 goldilocks_gen_tables_SOURCES = utils.c \
 					   goldilocks_gen_tables.c \
-					   arch_x86_64/f_impl.c \
+					   $(ARCH_DIR)/f_impl.c \
 	       			   f_arithmetic.c \
 	       			   f_generic.c \
 	      			   goldilocks.c \
@@ -22,7 +22,7 @@ lib_LTLIBRARIES = libgoldilocks.la
 libgoldilocks_la_SOURCES = utils.c \
 		      shake.c \
 		      spongerng.c \
-		      arch_x86_64/f_impl.c \
+		      $(ARCH_DIR)/f_impl.c \
 		      f_arithmetic.c \
 		      f_generic.c \
 		      goldilocks.c \
